gen/archExtra_gen.ml

```diff
@@ -257,6 +260,19 @@ and type special3 = I.special3
       friends = RegMap.empty;
       next_addr = 0; }
 
+  let used_register init =
+    List.filter_map ( function
+      | (Reg (_,r),Some (S _)) -> Some r
```

Comment:
IIUC, used register creates a list of registers already used in the init list. Why do you exclude registers that are initialised to values other than `S`? For example, PTE values (`AArch64PteVal.t`)

Reply:
I made a reserve decision here to not involved in `PTE` value. Let me explore how to generalise to also for `PTE` value. I feel it should be simple. The idea here is those used registers will be removed from the allocation pool and other thread will just reuse the same register without much check. I need to ensure these greedy way does not lead to problem.

Reply:
I update the PR, so it will assign the same register to the same PTE value cross process/thread as well.